pkg/iac/terraform/state/backend/http_reader.go

```diff
@@ -17,7 +17,11 @@ type HTTPBackend struct {
 	reader io.ReadCloser
 }
 
-func NewHTTPReader(rawURL string, opts *Options) (*HTTPBackend, error) {
+type HttpClient interface {
```

Comment:
This should be put in `test/http.go` with the mock file alongside (mockery default config)

Reply:
That'd make the HttpClient become an interface for testing, but it's actually used in the backend package. I don't think we want to use a test interface in our code : `func NewHTTPReader(client test.HttpClient...`, do we?

Reply:
That's true, the fact is that this generic interface should not be in `pkg/iac/terraform/state/backend/`.

Maybe we can move it to a new `pkg/http` package ?